Remotion/ObjectBinding/Sample/XmlReflectionBusinessObjectStorageProvider.cs

```diff
@@ -75,18 +70,20 @@ public BindableXmlObject GetObject (Type type, Guid id)
       if (obj != null)
         return obj;
 
+      lock (_reflectionBusinessObjectStorageProvider)
+      {
+        using (var stream = _reflectionBusinessObjectStorageProvider.GetReadObjectStream (type, id))
+        {
+          if (stream == null)
+            return null;
 
-      string typeDir = Path.Combine (_rootPath, type.FullName);
-      string fileName = Path.Combine (typeDir, id.ToString());
-      if (!File.Exists (fileName))
-        return null;
+          Type concreteType = GetConcreteType (type);
+          XmlSerializer serializer = GetXmlSerializer (concreteType);
 
-      Type concreteType = GetConcreteType (type);
-      XmlSerializer serializer = GetXmlSerializer (concreteType);
-      using (FileStream stream = new FileStream (fileName, FileMode.Open, FileAccess.Read))
-      {
-        obj = (BindableXmlObject) serializer.Deserialize (stream);
+          obj = (BindableXmlObject) serializer.Deserialize (stream);
+        }
       }
+
       obj._id = id;
       AddToIdentityMap (obj);
```

Comment:
The lock needs to cover `GetFromidentityMap` and `AddToIdentityMap` as well. that's why we also have to update `CreateObject`